rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1346,34 +1377,65 @@ public BooleanWrapper createNodeSource(String nodeSourceName, String infrastruct
         }
     }
 
-    protected BooleanWrapper createNodeSource(NodeSourceData nodeSourceData, boolean nodesRecoverable) {
-        String nodeSourceName = nodeSourceData.getName();
+    private String getUndeployedDescription(String infrastructureType, String policyType) {
+        String[] infraSplit = infrastructureType.split("\\.");
+        String simpleInfrastructureType = infraSplit[infraSplit.length - 1];
+        String[] policySplit = policyType.split("\\.");
+        String simplePolicyType = policySplit[infraSplit.length - 1];
+        return "Undeployed, Infrastructure: " + simpleInfrastructureType + ", Policy: " + simplePolicyType;
+    }
+
+    /**
+     * @deprecated  As of release 7.37, replaced by {@link #defineNodeSource(String, String, Object[], String, Object[],
+     * boolean)} and {@link #deployNodeSource(String)}
+     */
+    @Deprecated
+    public BooleanWrapper createNodeSource(String nodeSourceName, String infrastructureType,
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)